dask/array/routines.py

```diff
@@ -659,20 +659,61 @@ def histogram(a, bins=None, range=None, normed=False, weights=None, density=None
     """
     Blocked variant of :func:`numpy.histogram`.
 
-    Follows the signature of :func:`numpy.histogram` exactly with the following
-    exceptions:
-
-    - Either an iterable specifying the ``bins`` or the number of ``bins``
-      and a ``range`` argument is required as computing ``min`` and ``max``
-      over blocked arrays is an expensive operation that must be performed
-      explicitly.
-
-    - ``weights`` must be a dask.array.Array with the same block structure
-      as ``a``.
+    Parameters
+    ----------
+    a : array_like
+        Input data. The histogram is computed over the flattened array.
+    bins : int or sequence of scalars or str, optional
+        Either an iterable specifying the ``bins`` or the number of ``bins``
+        and a ``range`` argument is required as computing ``min`` and ``max``
+        over blocked arrays is an expensive operation that must be performed
+        explicitly.
+        If `bins` is an int, it defines the number of equal-width
+        bins in the given range (10, by default). If `bins` is a
+        sequence, it defines a monotonically increasing array of bin edges,
+        including the rightmost edge, allowing for non-uniform bin widths.
+        .. versionadded:: 1.11.0
+        If `bins` is a string, it defines the method used to calculate the
+        optimal bin width, as defined by `histogram_bin_edges`.
+    range : (float, float), optional
+        The lower and upper range of the bins.  If not provided, range
+        is simply ``(a.min(), a.max())``.  Values outside the range are
+        ignored. The first element of the range must be less than or
+        equal to the second. `range` affects the automatic bin
+        computation as well. While bin width is computed to be optimal
+        based on the actual data within `range`, the bin count will fill
+        the entire range including portions containing no data.
+    normed : bool, optional
+        .. deprecated:: 1.6.0
+        This is equivalent to the `density` argument, but produces incorrect
+        results for unequal bin widths. It should not be used.
+        .. versionchanged:: 1.15.0
+            DeprecationWarnings are actually emitted.
```

Comment:
I think we can remove the deprecated and version changed version references, since these refer to `numpy` versions and not `dask` versions (and might be a bit misleading).  I would change this to:

```suggestion
        This is equivalent to the ``density`` argument, but produces incorrect
        results for unequal bin widths. It should not be used.
```